pdl_migration.md

```diff
@@ -0,0 +1,255 @@
+---
+layout: guide
+title: Migrating from PDSC to PDL
+permalink: /pdl_migration
+excerpt: Guide for migrating from PDSC to PDL with a side-by-side comparison showing notable differences.
+---
+
+# Migrating from PDSC to PDL
+
+{{ page.excerpt }}
+
+## Contents
+- [Why Use PDL?](#why-use-pdl)
+- [How to Convert Schemas to PDL](#how-to-convert-schemas-to-pdl)
+- [Notable Differences Between PDSC and PDL](#notable-differences-between-pdsc-and-pdl)
+    - [PDL is More Java-Like](#pdl-is-more-java-like)
+    - [Import Statements](#import-statements)
+    - [Shorthand for Custom Properties](#shorthand-for-custom-properties)
+
+## Why Use PDL?
+
+PDL as a schema definition language was designed to be read and written by humans, making it a much more enjoyable experience for developers
+to define their service's models. PDSC, despite its syntax being simply a subset of JSON, cannot boast this human-readability factor.
+
+In addition to its inherently more readable syntax, PDL also features some extra shorthand which developers can leverage to write less and
+make their schemas more readable for others. See [below](#notable-differences-between-pdsc-and-pdl) for more information on this.
+
+## How to Convert Schemas to PDL
+
+If you have an existing project with PDSC schemas, converting everything to PDL is pretty straightforward.
+Rest.li's Gradle plugin provides a task `convert<sourceSet>ToPdl` which will automatically convert your schemas for you.
+It should be noted that this task verifies the converted schemas against the original schemas.
+If this verification is failed, then the whole conversion will be aborted.
+
+The following command will convert every PDSC schema in your project to PDL.
+
+```
+gradle convertToPdl
+```
+
+You can selectively migrate only one particular module of your project with the following:
+
+```
+gradle :<moduleName>:convertToPdl
+```
+
+This task also takes in a few options:
+
+```
+gradle :<moduleName>:convertToPdl \
+    [-PconvertToPdl.reverse=(true|false)] \
+    [-PconvertToPdl.keepOriginal=(true|false)] \
+    [-PconvertToPdl.preserveSrcCmd]
+```
+
+| Property | Type | Description
+|----------|------|------------
+| `convertToPdl.reverse` | boolean | If true, converts PDL schemas back to PDSC (and vice versa if false).
+| `convertToPdl.keepOriginal` | boolean | If true, keeps the source schemas (the source schemas are deleted by default).
+| `convertToPdl.preserveSrcCmd` | string | Command which is run for each file, useful for running special VCS logic. The command should be a template string containing `$src` and `$dst` as references to the source and destination filename, respectively.
```

Comment:
I think we should provide an example here, like - gradle  convertToPdl -PconvertTodl.preserveSrcCmd "/usr/bin/svn mv \$src \$dst"